### DISCUSSION TITLE
How to set suitable time intervals for different controllers?

### DISCUSSION THREAD

**Discussion by frank-bee** [fluxcd/flux2#1076] [Q&A]
I'm asking myself how to set the _interval_ values.
Are the controllers running completely asynchronously?
E.g. for HelmRelease: It  references a GitSource ad as well  Kustomize resource is involved . 
How to the 3 time intervals play together?
Shouldn't it be enough to set a interval for the GitSource and with any change the other controllers (kustomize and helm) are triggered implicitely?

My intention is to not overload the Git source access because I still face problems sometimes when accessing Azure devops repos ( flaky behaviour, error log entries several times a day)

**Comment by hiddeco** [comment]
They run completely asynchronously but may watch for changes to objects reconciled by another controller.

**Concrete examples:**

`GitRepository` with an interval of `10m` will check the upstream Git repository every 10 minutes for changes.

`HelmChart` with an interval of `15m` will check the _artifact_ of the `GitRepository` every 15 minutes, but as it also watches the `GitRepository` for changes, it will get reconciled when a new `GitRepository` artifact revision becomes available.

`HelmRelease` with an interval of `30m` will check the _artifact_ of the `HelmChart` every 30 minutes (and reconcile Helm state), but if a new `HelmChart` revision is made available, it will get reconciled right away.

**Reply by frank-bee** [reply]
Thanks for the explanations @hiddeco !

So if I have a `HelmRelease` pointing to a `HelmChart` via `GitRepository` source ref, then only the interval of the `GitRepository` is of relevance, correct?

E.g.

```   
chart:
    spec:
      chart: chart
      sourceRef:
        kind: GitRepository
        name: foo
  ```

**Reply by frank-bee** [reply]
If so , I would omit the interval value for the `HelmRelease` , better not explicitly defining it in this case to avoid confusion

**Reply by hiddeco** [reply]
No, it is not as the controller does a lot more than just checking if there is a new chart available. It for example also confirms the Helm storage has not been changed, and the release still exists.

If you would omit the `interval` (which is not actually possible, as this is a required field), all those things would no longer be checked and acted upon.

**Reply by frank-bee** [reply]
Got it , thanks @hiddeco for the insides!
